### PR TITLE
Create power_of_two.dart

### PR DESCRIPTION
**Pull request by neha-hasija17**
This pull request is for HacktoberFest 2020
In this I have checked if a number is a power of 2 or not, without using logarithmic operations and by using bit manipulation.

# Welcome to Dart community

### **Describe your change:**

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [ ] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Dart/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Dart files are placed inside an existing directory.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: maths/power_of_two.dart [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+}
+
+void main() {
+  power_of_two(10);
+  power_of_two(23);
+  power_of_two(32);
+  power_of_two(2234);
+  power_of_two(2048);
+
+  return;
```
Comment (Parowicz) [inline maths/power_of_two.dart:16]: This `return` is redundant.
```suggestion
```

**Comment by neha-hasija17** [comment]
Thankyou sir for reviewing my program. Please spare some time and recorrect
the error in tests.

File: maths/power_of_two.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+void power_of_two(int n) {
+  int result = n & (n - 1);
+  if (result == 0)
+    print("Yes, the number $n is a power of 2");
+  else
+    print("No, the number $n is not a power of 2");
+}
```
Comment (Parowicz) [inline maths/power_of_two.dart:7]:
```suggestion
import 'package:test/test.dart';

bool power_of_two(int n) {
  return (n & (n - 1)) == 0;
}
```

File: maths/power_of_two.dart [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+void main() {
+  power_of_two(10);
+  power_of_two(23);
+  power_of_two(32);
+  power_of_two(2234);
+  power_of_two(2048);
```
Comment (Parowicz) [inline maths/power_of_two.dart:14]: Hmm, it's returning `true` for `0`. 
```suggestion
  test("Test power_of_two 0 returns false", () {
    expect(power_of_two(0), isFalse);
  });

  test("Test power_of_two 1 returns true", () {
    expect(power_of_two(1), isTrue);
  });

  test("Test power_of_two 10 returns false", () {
    expect(power_of_two(10), isFalse);
  });

  test("Test power_of_two 10 returns false", () {
    expect(power_of_two(10), isFalse);
  });

  test("Test power_of_two 23 returns false", () {
    expect(power_of_two(23), isFalse);
  });

  test("Test power_of_two 32 returns true", () {
    expect(power_of_two(32), isTrue);
  });

  test("Test power_of_two 2234 returns false", () {
    expect(power_of_two(2234), isFalse);
  });

  test("Test power_of_two 2048 returns true", () {
    expect(power_of_two(2048), isTrue);
  });
```

**Comment by neha-hasija17** [comment]
@Parowicz please review this

**Comment by neha-hasija17** [comment]
@Parowicz sir please spare some time and have a look

**Comment by Parowicz** [comment]
Hello, during first review I've noticed that `power_of_two` returns `true` for `0`.
Is it correct?

**Comment by neha-hasija17** [comment]
Hello sir, Actually for 0 it returns false.

**Comment by neha-hasija17** [comment]
@Parowicz kindly review this

**Comment by neha-hasija17** [comment]
@Parowicz please remove the label of tests are needed from this PR. :)

**Comment by Parowicz** [comment]
Looks Great! Thank you!

**Pull request merged**